docs/source/examples/vqe-demo_custom-gate.myst

```diff
@@ -0,0 +1,282 @@
+    for j, term in enumerate(pauli_sum.terms):
+        qubits_to_measure = []
+        for index, gate in term:
+            qubits_to_measure.append(index)
+            meas_outcome = expectation_from_sampling(
+                thetas, noisy, qubits_to_measure, backend, samples
+            )
+            expectation += term.coefficient * meas_outcome
+    return expectation.real
+```
+
+The following function is a modified version of `expectation_from_sampling' 
+from https://github.com/rigetti/grove/blob/master/grove/pyvqe/vqe.py. 
+It is modified to follow pyQuil conventions for defining custom gates.
+
+```{code-cell} ipython3
+def expectation_from_sampling(
+    thetas, executable: Program, marked_qubits: List[int], backend, samples: int
+) -> float:
+    """Calculate the expectation value of the Zi operator where i ranges over all 
+    qubits given in marked_qubits.
+    """
+    bitstring_samples = backend.run(executable, memory_map={"theta": thetas})
+    bitstring_tuples = list(map(tuple, bitstring_samples))
+
+    freq = Counter(bitstring_tuples)
+
+    
+    exp_val = 0
+    for bitstring, count in freq.items():
+        bitstring_int = int("".join([str(x) for x in bitstring[::-1]]), 2)
+        if parity_even_p(bitstring_int, marked_qubits):
+            exp_val += float(count) / samples
+        else:
+            exp_val -= float(count) / samples
+    return exp_val
+```
+
+Calculate the parity of elements at indexes in marked_qubits
+The function is a modified version of `parity_even_p` from 
+https://github.com/rigetti/grove/blob/master/grove/pyvqe/vqe.py.
+
+```{code-cell} ipython3
+def parity_even_p(state, marked_qubits):
+    mask = 0
+    for q in marked_qubits:
+        mask |= 1 << q
+    return bin(mask & state).count("1") % 2 == 0
+```
+
+Scan over the parameter theta and plot energy expecation, without mitigation
+
+```{code-cell} ipython3
+thetas = np.linspace(0, 2 * np.pi, 51)
+results = []
+
+for theta in thetas:
+    results.append(executor(theta, backend, readout_qubit, samples, hamiltonian, program))
+```
+
+Run VQE routine without noise mitigation
+
+```{code-cell} ipython3
+init_angle = [3.0]
+
+res = optimize.minimize(
+    executor,
+    init_angle,
+    args=(backend, readout_qubit, samples, hamiltonian, program),
+    method="Nelder-Mead",
+    options={"xatol": 1.0e-3, "fatol": 1.0e-2},
+)
+print(res)
+```
+
+Now introduce noise mitigation technique Zero Noise Extrapolation
+
+Create new objective function to calculate expecation value w/ noise mitigation
+
+```{code-cell} ipython3
+def mitigated_expectation(
+    thetas, backend, readout_qubit, samples, pauli_sum, executable: Program, factory
+) -> float:
+    mitigated_exp = zne.execute_with_zne(
+        executable,
+        lambda p: executor(thetas, backend, readout_qubit, samples, pauli_sum, p),
+        factory,
+        fold_gates_from_right,
+    )
+    return mitigated_exp
+```
+
+Here we use a linear technique for the Zero Noise Extrapolation
```

Comment:
You could add a link to the relevant part in the Users Guide where linear factories are discussed, e.g., https://mitiq.readthedocs.io/en/latest/guide/guide-zne.html#classical-fitting-and-extrapolation-factory-objects

Reply:
Agree, added.